src/daft-core/src/array/ops/any_value.rs

```diff
@@ -0,0 +1,104 @@
+use arrow2::{array::PrimitiveArray, bitmap::Bitmap};
+use common_error::DaftResult;
+
+use crate::{
+    array::{DataArray, FixedSizeListArray, ListArray, StructArray},
+    datatypes::*,
+};
+
+use super::{full::FullNull, DaftAnyValueAggable, GroupIndices};
+
+fn get_any_grouped_idx(
+    groups: &GroupIndices,
+    ignore_nulls: bool,
+    validity: Option<&Bitmap>,
+) -> UInt64Array {
+    let group_indices = if ignore_nulls && let Some(validity) = validity {
+        Box::new(PrimitiveArray::from_trusted_len_iter(groups.iter().map(|g| {
+            for i in g {
+                if validity.get_bit(*i as usize) {
+                    return Some(*i)
+                }
+            }
+            None
+        })))
+    } else {
+        Box::new(PrimitiveArray::from_trusted_len_iter(groups.iter().map(|g| g.first().cloned())))
+    };
+
+    DataArray::from(("", group_indices))
+}
+
+impl<T> DaftAnyValueAggable for DataArray<T>
+where
+    T: DaftNumericType,
+{
+    type Output = DaftResult<DataArray<T>>;
+
+    fn any_value(&self, ignore_nulls: bool) -> Self::Output {
+        if ignore_nulls && let Some(validity) = self.validity() {
+            for i in 0..self.len() {
```

Comment:
you can rewrite this as `validity.iter().position(|i| validity.get_bit)`

https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.position